Branch-SDK/src/main/java/io/branch/referral/ServerRequestCreateUrl.java

```diff
@@ -88,11 +85,15 @@ public ServerRequestCreateUrl(Context context, final String alias, final int typ
 
             setPost(linkPost_);
 
+            //Remove unnecessary fields
+            linkPost_.remove("anon_id");
```

Comment:
Where are these added from originally? I think i'd rather have them omitted to begin with than exist on the object indeterminately. 

I don't want to create feature scope but it's a good idea to be additive than subtractive wrt only requiring we need to begin with.

Reply:
They're added in the setPost method above, which calls DeviceInfo's updateRequestWithV1Params. Since that function just adds all of the fields to each request, the alternative would be to add some checks in that function for specific request types.

Reply:
Yeah we'll want to refactor that so that each type is explicit about its requirements. (in another pr)